lib/internal/loader/Loader.js

```diff
@@ -0,0 +1,76 @@
+'use strict';
+
+const URL = require('url').URL;
+
+const ModuleWrap = process.binding('module_wrap').ModuleWrap;
+
+const ModuleMap = require('internal/loader/ModuleMap');
+const ModuleJob = require('internal/loader/ModuleJob');
+const resolveRequestUrl = require('internal/loader/resolveRequestUrl');
+const {Error} = global;
+
+const pathToFileURL = (pathname) => `file://${pathname}/`;
+
+const getNamespaceOf = (m) => {
+  const tmp = new ModuleWrap('import * as _ from "";_;', '');
+  tmp.link(async () => m);
+  tmp.instantiate();
+  return tmp.evaluate();
+};
+
+function getBase() {
+  try {
+    return pathToFileURL(process.cwd());
+  } catch (e) {
+    // If the current working directory no longer exists.
+    if (e.code === 'ENOENT') {
+      return undefined;
+    }
+    throw e;
+  }
+}
+
+class Loader {
+  constructor(base = getBase()) {
+    this.moduleMap = new ModuleMap();
+    this.base = base ? new URL(`${base}`) : base;
```

Comment:
See above, this is necessary to keep Loader construction from throwing, in a case when it's still perfectly fine to use native modules and absolute paths.